internal/device/energy_zone.go

```diff
@@ -0,0 +1,154 @@
+func NewAggregatedZone(zones []EnergyZone) *AggregatedZone {
+	// Panic on invalid inputs
+	if len(zones) == 0 {
+		panic("NewAggregatedZone: zones cannot be empty")
+	}
+
+	// Use the first zone's name as the aggregated zone name
+	name := zones[0].Name()
+	// Calculate and cache the combined MaxEnergy during construction
+	// Check for overflow when summing MaxEnergy values
+	var totalMax Energy
+	for _, zone := range zones {
+		zoneMax := zone.MaxEnergy()
+		// Check for overflow before adding
+		if totalMax > 0 && zoneMax > math.MaxUint64-totalMax {
+			// Overflow would occur, use MaxUint64 as safe maximum
+			totalMax = Energy(math.MaxUint64)
+			break
+		}
+		totalMax += zoneMax
+	}
+
+	return &AggregatedZone{
+		name:          name,
+		index:         -1, // Indicates this is an aggregated zone
+		zones:         zones,
+		lastReadings:  make(map[zoneKey]Energy),
+		currentEnergy: 0,
+		maxEnergy:     totalMax, // Cache the combined MaxEnergy
+	}
+}
+
+// Name returns the zone name
+func (az *AggregatedZone) Name() string {
+	return az.name
+}
+
+// Index returns the zone index (-1 for aggregated zones)
+func (az *AggregatedZone) Index() int {
+	return az.index
+}
+
+// Path returns path for the aggregated zone
+func (az *AggregatedZone) Path() string {
+	// TODO: decide if all the paths should be returned
+	return fmt.Sprintf("aggregated-%s", az.name)
+}
+
+// Energy returns the total energy consumption across all aggregated zones,
+// handling wrap-around for each individual zone
+func (az *AggregatedZone) Energy() (Energy, error) {
+	az.mu.Lock()
+	defer az.mu.Unlock()
+
+	var totalDelta Energy
+
+	for _, zone := range az.zones {
+		currentReading, err := zone.Energy()
+		if err != nil {
+			return 0, fmt.Errorf("no valid energy readings from aggregated zones - %s: %w", zone.Name(), err)
+		}
+
+		zoneID := zoneKey{zone.Name(), zone.Index()}
+
+		if lastReading, exists := az.lastReadings[zoneID]; exists {
+
+			// Calculate delta since last reading
+			var delta Energy
+			if currentReading >= lastReading {
+				// Normal case: no wrap
+				delta = currentReading - lastReading
+			} else {
+				// Wrap occurred: calculate delta across wrap boundary
+				// Only if zone has valid MaxEnergy (> 0)
+				if zone.MaxEnergy() > 0 {
+					delta = (zone.MaxEnergy() - lastReading) + currentReading
+				} else {
+					// Invalid MaxEnergy, treat as normal delta (might be negative)
+					delta = currentReading - lastReading
+				}
+			}
+			totalDelta += delta
+		} else {
+			// First reading: use current reading as initial energy
+			totalDelta += currentReading
+		}
+
+		// Update last reading
+		az.lastReadings[zoneID] = currentReading
+	}
+
+	// Update aggregated energy counter
+	az.currentEnergy += totalDelta
+
+	// Wrap at maxEnergy boundary to match hardware counter behavior
+	// This is required for the power attribution algorithm's calculateEnergyDelta()
+	if az.maxEnergy > 0 {
+		az.currentEnergy %= az.maxEnergy
+	}
```

Comment:
what is  the need for this wraparound when totalMax already takes care of overflow

Reply:
I think it may not be necessary but let me check that again .. for now this shouldn't change how the aggregated zone should behave. 
Adding an issue